### PR TITLE
[PT-BR] Import Export.ts, JSDoc Support.ts and Generic Classes.ts

### PR DESCRIPTION
**Pull request by rafaelrozon**
Tagging @khaosdoctor, @alvarocamillont, @danilofuchs e @orta  as instructed on https://github.com/microsoft/TypeScript-Website/issues/233.

I have a couple of questions that I would like to clarify:

1. Should the sample code in the Playground docs also be translated or it should remain in English?
2. There're some terms that I'm not sure how to translate and if there is a translation more common in Portuguese. Do you guys follow some convention for technical terms or have a list of technical terms translation from English to Portuguese?

I added a `// !!!!` to where I need to review the translation for some terms. I'll remove those when everyone is happy with the translation.

I'm not sure about these terms:
- named import/export
- logging
- default export
- built-in
- auto-complete

Thank you!

### CODE REVIEW

**Comment by orta** [comment]
1, I generally recommend translating the sample code so that it makes sense to someone without _a lot_ of English context
2, we don't, I'm open to someone making a word list, but it's not something _I_ can do

**Comment by khaosdoctor** [comment]
About the questions, 1 is like @orta said. We generally recommend the translation of code files because they create context.

On 2, I'll answer all the words here since we still don't have a word list, that's an idea but I hardly think this would be used.

- named import/export => import/export nomeado
- logging -> logging
- default export -> export padrão
- built-in -> nativo
- auto-complete -> auto-complete

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+//// { order: 1, target: "ES5" }
+
+// JavaScript adicionou import/export à linguage em 2016
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// JavaScript adicionou import/export à linguagem em 2016
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+// e TypeScript tem suporte complete para essee estilo de
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// e TypeScript tem suporte completo para esse estilo de
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+// conexão entre arquivos e para com modulos externos.
+// TypeScript expande essa sintaxe ao permitir que tipos
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// O TypeScript expande essa sintaxe ao permitir que tipos
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+// Nomeando especificamente quais importações você precisa,
+// permite que ferramentas tenham a habilidade de remover
+// código não utilizado em seu applicativo, e ajuda você a
+// entender o que está sendo utilizado nessa arquivo em particular.
+
+// Nesse caso: danger, message e warn são importações de JavaScript
+// enquanto que DangerDSLType é uma interface.
+
+// TypeScript permite que engenheiros documentem seus códigos usando
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// O TypeScript permite que engenheiros documentem seus códigos usando
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+// JSDoc, e documentos são importados também. Por exemplo, se você
+// passar o mouse sobre diferentes partes abaixo, você verá
+// explicações sobre o que elas são.
+
+danger.git.modified_files;
+
+// Se você quiser saber como criar essa anotações de documentação
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// Se você quiser saber como criar essas anotações de documentação
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+// leia o exemplo:jsdoc-support
+
+// !!!!!
+// Outro modo de importar código é através da exportação
+// padråo de um módulo. Um exemplo disso é o módulo debug, o qual
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// padrão de um módulo. Um exemplo disso é o módulo debug, o qual
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+// JSDoc, e documentos são importados também. Por exemplo, se você
+// passar o mouse sobre diferentes partes abaixo, você verá
+// explicações sobre o que elas são.
+
+danger.git.modified_files;
+
+// Se você quiser saber como criar essa anotações de documentação
+// leia o exemplo:jsdoc-support
+
+// !!!!!
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion

```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+// Outro modo de importar código é através da exportação
+// padråo de um módulo. Um exemplo disso é o módulo debug, o qual
+// expõe uma funcão que cria uma função de log.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// expõe uma função que cria uma função de log.
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+import debug from "debug";
+const log = debug("playground");
+log("Started running code");
+
+// !!!!!
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion

```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+const stickerGenerator = () => {};
+export default stickerGenerator;
+
+// Isto poderia ser importado em outro arquivo usando:
+//
+// import getStickers from "./path/to/file"
+//
+// O nome é decidido pelo módulo que está realizando a importação.
+
+// Estes não são os únicoss tipos de importação, apenas os
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// Estes não são os únicos tipos de importação, apenas os
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+//
+// https://www.typescriptlang.org/docs/handbook/modules.html
+
+// Entretanto, para tentar responder a última questão. Se você
+// observar o código JavaScript desse exemplo, você verá isto:
+
+// var stickerGenerator = function () { };
+// exports.default = stickerGenerator;
+
+// Isso define a propriedade default no objeto exports como
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// Isso define a propriedade padrão no objeto exports como
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+// stickerGenerator. Existe código que define o exports como uma
+// função ao invés de um objeto.
+//
+// TypeScript optou por utilizar a especificação ECMAScript
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// O TypeScript optou por utilizar a especificação ECMAScript
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+// sobre como lidar com esses casos, que é criar um error.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// sobre como lidar com esses casos, que é criar um erro.
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+// Entretanto, existe uma configuração do compilador que lida
+// automaticamente com esses casos para você, a qual é a esModuleInterop.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// automaticamente com esses casos para você, que é a esModuleInterop.
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+//
+// Se você ativar ela nessa exemplo, você verá que o erro desaparece.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]:
```suggestion
// Se você ativar ela nesse exemplo, você verá que o erro desaparece.
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
+//// { order: 3, isJavaScript: true }
+
+// TypeScript tem um suporte muito rico de JSDoc, para muitos casos
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// TypeScript tem um suporte muito rico ao JSDoc, para muitos casos
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+// você pode até não utilizar arquivos .ts e apenas usar annotações de JSDoc
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// você pode até não utilizar arquivos .ts e apenas usar anotações de JSDoc
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+// para criar um ambiente de desenvolvimento rico.
+//
+// Um comentário JSDoc é um comentário de multiplas linhas que
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// Um comentário JSDoc é um comentário de múltiplas linhas que 
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+
+// Comentários JSDoc pertencem ao código JavaScript mais 
+// próximo abaixo dele
+
+const myVariable = "Hi";
+
+// Se você passar o mouse sobre myVariable, você pode ver que 
+// ela tem o texto do comentário JSDoc acima.
+
+// Comentáios JSDoc são um meio de fornecer informação sobre os tipos utilizados
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// Comentários JSDoc são um meio de fornecer informação sobre os tipos utilizados
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+// para o TypesScript e editores. Vamos começar com um fácil,
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// para o TypeScript e editores. Vamos começar com um fácil,
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+// definindo o tipo de uma variável como o tipo padrão. 
+
+// !!!!
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion

```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+  return a + b;
+}
+
+// Você pode usar a maioria das ferramentas de tipos do TypeScript, como uniões (unions):
+
+/** @type {(string | boolean)} */
+let stringOrBoolean = "";
+stringOrBoolean = false;
+
+// Extendendo globais em JSDoc é um processo mais envolvente
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// Estender globais em JSDoc é um processo mais complicado
```

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+// o qual você pode conferir no manual do VS Code:
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]:
```suggestion
// que você pode conferir no manual do VS Code:
```

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts [hunk 24]
```diff
@@ -0,0 +1,95 @@
+//// { order: 3 }
+
+// Este exemplo é em sua maioria em TypeScript, porque é um modo muito
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]:
```suggestion
// Este exemplo é, em sua maioria, em TypeScript, porque é um modo muito
```

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+  adicionar(objeto: TipoDeRoupa) {
+    this.conteudo.push(objeto);
+  }
+
+  remover() {
+    return this.conteudo.pop();
+  }
+}
+
+// Para user uma Gaveta você precisará trabalhar com outro tipo:
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]:
```suggestion
// Para usar uma Gaveta você precisará trabalhar com outro tipo:
```

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+gavetaDeCamisetas.adicionar({ tamanho: "m" });
+
+// Se você for um pouco excêntrico, você pode até criar
+// uma gaveta que tem Meias e Camisetas usando para isso
+// uma união (union):
+
+const gavetaMista = new Gaveta<Meia | Camiseta>();
+
+// Criar uma classe como a Gaveta sem a sintaxe do TypeScript
+// requer o uso da etiqueta chamada template (template tag) em JSDoc.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]:
```suggestion
// requer o uso da tag "template" em JSDoc.
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]
Small changes

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts (hunk 26 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts (hunk 25 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts (hunk 24 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Working With Classes/Generic Classes.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 23 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 22 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 21 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 20 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 19 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 18 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 17 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js (hunk 16 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/JSDoc Support.js]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 15 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 14 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 13 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 12 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 11 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 10 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 9 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 8 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 6 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 7 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 5 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 4 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 3 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 2 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

File: packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts (hunk 1 above)
Comment (rafaelrozon) [reply packages/playground-examples/copy/pt/JavaScript/Modern JavaScript/Import Export.ts]: Fixed.

**Review by rafaelrozon — COMMENTED** [review]
Thank you @khaosdoctor and @orta  for the review. I addressed all the comments and did some small changes. Let me know if there's anything else that needs a fix. 

Thanks!

**Comment by khaosdoctor** [comment]
I guess that'd be all! Thanks for the PR :D

**Review by khaosdoctor — APPROVED** [review]

**Comment by khaosdoctor** [comment]
lgtm

**Comment by rafaelrozon** [comment]
Thanks, Lucas and Orta for reviewing it.

**Pull request merged**